src/lib/crypto/builtin/sha2/sha256.c

```diff
@@ -211,7 +211,7 @@ k5_sha256_update(SHA256_CTX *m, const void *v, size_t len)
 #if !defined(WORDS_BIGENDIAN) || defined(_CRAY)
 	    int i;
 	    uint32_t current[16];
-	    struct x32 *u = (struct x32*)m->save;
+	    struct x32 *u = (struct x32*)(void*)m->save;
```

Comment:
There's another branch down below which needs similar treatment.  (Also for the next hunk.)

There might be a more correct way to express this code (using a different type for save), but I'm reluctant to diverge materially from the upstream Heimdal code.  I suppose there might be different upstream sha256/sha512 code we could think about using instead.